portals/publisher/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx

```diff
@@ -242,9 +252,36 @@ function EndpointOverview(props) {
         return supportedEndpointTypes;
     };
 
+    /**
+     * Retrieve the service list from the service catalog
+     */
+    function getServices() {
```

Comment:
Let's make this a custom hook, For example: https://github.com/wso2-enterprise/choreo-console/blob/main/src/hooks/marketplace.tsx#L167-L193

We need to handle 3 conditions for all API calls

1. Response data
2. API loading state (show a loader while API call is in progress)
3. Handle errors (catch network errors and show a meaningful message to user )

Reply:
To add these have to change all the current API call formats in the service catalog. Since it has a big change I will do it after the alpha.